data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorConfig.java

```diff
@@ -35,87 +37,159 @@ public class KeyValueProcessorConfig {
+    @JsonPropertyDescription("Specifies whether to group values using predefined value grouping delimiters: " +
+            "`{...}`, `[...]`, `<...>`, `(...)`, `\"...\"`, `'...'`, `http://... (space)`, and `https:// (space)`. " +
+            "If this flag is enabled, then the content between the delimiters is considered to be one entity and " +
+            "is not parsed for key-value pairs. Default is `false`. If `value_grouping` is `true`, then " +
+            "`{\"key1=[a=b,c=d]&key2=value2\"}` parses to `{\"key1\": \"[a=b,c=d]\", \"key2\": \"value2\"}`.")
     private boolean valueGrouping = DEFAULT_VALUE_GROUPING;
 
     @JsonProperty("recursive")
+    @JsonPropertyDescription("Specifies whether to recursively obtain additional key-value pairs from values. " +
+            "The extra key-value pairs will be stored as sub-keys of the root key. Default is `false`. " +
+            "The levels of recursive parsing must be defined by different brackets for each level: " +
+            "`[]`, `()`, and `<>`, in this order. Any other configurations specified will only be applied " +
+            "to the outmost keys.\n" +
+            "When `recursive` is `true`:\n" +
+            "`remove_brackets` cannot also be `true`;\n" +
+            "`skip_duplicate_values` will always be `true`;\n" +
+            "`whitespace` will always be `\"strict\"`.")
     @NotNull
     private boolean recursive = DEFAULT_RECURSIVE;
 
     @JsonProperty("tags_on_failure")
+    @JsonPropertyDescription("When a `kv` operation causes a runtime exception within the processor, " +
+            "the operation is safely stopped without crashing the processor, and the event is tagged " +
+            "with the provided tags.")
     private List<String> tagsOnFailure;
 
     @JsonProperty("overwrite_if_destination_exists")
+    @JsonPropertyDescription("Specifies whether to overwrite existing fields if there are key conflicts " +
+            "when writing parsed fields to the event. Default is `true`.")
     private boolean overwriteIfDestinationExists = true;
 
     @JsonProperty("drop_keys_with_no_value")
+    @JsonPropertyDescription("Specifies whether keys should be dropped if they have a null value. Default is `false`. " +
+            "If `drop_keys_with_no_value` is set to `true`, " +
+            "then `{\"key1=value1&key2\"}` parses to `{\"key1\": \"value1\"}`.")
     private boolean dropKeysWithNoValue = false;
 
     @JsonProperty("key_value_when")
+    @JsonPropertyDescription("Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), " +
+            "such as `/some-key == \"test\"`, that will be evaluated to determine whether " +
+            "the processor should be applied to the event.")
     private String keyValueWhen;
 
     @JsonProperty("strict_grouping")
+    @JsonPropertyDescription("Specifies whether strict grouping should be enabled when the `value_grouping` " +
```

Comment:
Same here. Let's include the text in the example column as well for this parameter.